src/main/kotlin/controller/Controller.kt

```diff
@@ -2,6 +2,7 @@ package controller
 
```

Comment:
I see that you've been working on step 2 as we were reviewing step 1. 👏 
That resulted in kind of a complex commit log.
Once you get some time, I recommend trying out `rebase` and `cherry-pick` features.
They are implemented in IntelliJ, so definitely check it out; it'll be easier to use than command line, especially when cherry-picking.

Oh and these features are used daily in actual work, so they're pretty important as well. :)